### PR TITLE
Added a generic way to translate the necessary fields in composite entities.

### PR DESCRIPTION
**Pull request by 2403905**
## Description

Added a generic way to translate the necessary fields in composite entities.
```
// TranslateEntity function provides the generic way to translate a struct, array or slice.
// Support for maps is also provided, but non-pointer values will not work.
// The function also takes the entity with fields to translate.
// The function supports nested structs and slices of structs.
/*
tr := NewTranslator("en", _domain, _fsys)
// a slice of translatables can	be passed directly
val := []string{"description", "display name"}
err := tr.TranslateEntity(tr, s, val)
// string maps work the same way
val := map[string]string{
	"entryOne": "description",
	"entryTwo": "display name",
}
err := TranslateEntity(tr, val)
// struct fields need to be specified
type Struct struct {
	Description string
	DisplayName string
	MetaInformation string
}
val := Struct{}
err := TranslateEntity(tr, val,
	l10n.TranslateField("Description"),
	l10n.TranslateField("DisplayName"),
)
// nested structures are supported
type InnerStruct struct {
	Description string
	Roles []string
}
type OuterStruct struct {
	DisplayName string
	First InnerStruct
	Others map[string]InnerStruct
}
val := OuterStruct{}
err := TranslateEntity(tr, val,
	l10n.TranslateField("DisplayName"),
	l10n.TranslateStruct("First",
		l10n.TranslateField("Description"),
		l10n.TranslateEach("Roles"),
	),
	l10n.TranslateMap("Others",
		l10n.TranslateField("Description"),
	},
*/
```
## Related Issue

- Fixes https://github.com/owncloud/ocis/issues/9700

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by 2403905** [comment]
@kobergj The translate functions are separated semantically 
https://github.com/2403905/ocis/compare/7013d7abf80693ee8aa4133b76339df0c8772995..e0b31d489d1acaf9613958beff5c03e4c1389598

File: ocis-pkg/l10n/l10n.go [hunk 1]
```diff
@@ -63,6 +69,61 @@ func (t Translator) Locale(locale string) *gotext.Locale {
...
+err := TranslateEntity(tr, val,
+	l10n.TranslateField("DisplayName"),
+	l10n.TranslateStruct("First",
+		l10n.TranslateField("Description"),
+		l10n.TranslateEach("Roles"),
+	),
+	l10n.TranslateMap("Others",
+		l10n.TranslateField("Description"),
+	},
+*/
```
Comment (butonic) [inline ocis-pkg/l10n/l10n.go:122]: why is all this commented?

File: ocis-pkg/l10n/l10n.go [hunk 2]
```diff
@@ -91,3 +152,194 @@ func GetUserLocale(ctx context.Context, userID string, vc settingssvc.ValueServi
...
+			case reflect.Slice, reflect.Array, reflect.Map:
+				translateEach(tr, v, args...)
+			}
+		}
+	case reflect.Map:
+		for _, k := range value.MapKeys() {
+			v := value.MapIndex(k)
+			switch v.Kind() {
+			case reflect.Struct:
+				// FIXME: add support for non-pointer values
```
Comment (butonic) [inline ocis-pkg/l10n/l10n.go:248]: in a subsequent pr?

**Review by butonic — APPROVED** [review]

File: ocis-pkg/l10n/l10n.go (hunk 1 above)
Comment (kobergj) [reply ocis-pkg/l10n/l10n.go:122]: info on how to use the pkg

**Pull request merged**